runtime/src/main/cpp/Worker.cpp

```diff
@@ -475,6 +492,25 @@ class State {
     }
   }
 
+  void waitNativeWorkerTerminationUnlocked(KInt id) {
+      pthread_t threadToWait = 0;
+      {
+          Locker locker(&lock_);
+
+          for (auto& kvp : terminating_native_workers_) {
+              RuntimeAssert(!pthread_equal(kvp.second, pthread_self()), "Native worker is joining with itself");
+              if (kvp.first == id) {
+                  threadToWait = kvp.second;
+                  break;
```

Comment:
`waitNativeWorkersTerminationUnlocked` above also clears `terminating_native_workers_`. Shouldn't we remove the worker from that registry here too?

Reply:
We should, fixed.